### ISSUE TITLE
Verify Realm.logger behavior across isolates

### ISSUE THREAD

**Issue by nirinchev** [realm/realm-dart#1446]
It looks like in several places we're calling `Realm.logger.log` directly which is likely to behave in unexpected ways when invoked on a background isolate. Additionally, setting `Realm.logger` on a background isolate is likely to overwrite the configured logger on the main isolate. We should warn about doing it and probably also route the messages through Core so that logs from the SDK get to the main isolate logger.

**Comment by nielsenko** [comment]
One thing to consider (if you haven't already) is that `Realm.logger` is a static, but in dart statics are not shared across isolates. It might be a bit surprising if you add a listener to an isolates `Realm.logger` but it never gets called, since the messages are routed to the main isolate only.

That said, I agree this is kind of a dark corner. I'm not sure what is best.

**Comment by nirinchev** [comment]
Yes, this was my concern precisely. I think there may be some trick we can employ to make it almost transparent for users, but haven't gotten around to try it out.
